test/java/com/arpnetworking/metrics/portal/alerts/impl/AlertJobRepositoryTest.java

```diff
@@ -0,0 +1,142 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.arpnetworking.metrics.portal.alerts.impl;
+
+import com.arpnetworking.metrics.portal.TestBeanFactory;
+import com.arpnetworking.metrics.portal.alerts.AlertRepository;
+import com.arpnetworking.metrics.portal.alerts.scheduling.AlertExecutionContext;
+import com.arpnetworking.metrics.portal.alerts.scheduling.AlertJobRepository;
+import com.arpnetworking.metrics.portal.scheduling.Schedule;
+import com.arpnetworking.metrics.portal.scheduling.impl.NeverSchedule;
+import com.google.common.collect.ImmutableList;
+import models.internal.AlertQuery;
+import models.internal.Organization;
+import models.internal.QueryResult;
+import models.internal.alerts.Alert;
+import models.internal.alerts.AlertEvaluationResult;
+import models.internal.impl.DefaultAlert;
+import models.internal.impl.DefaultAlertQuery;
+import models.internal.impl.DefaultMetricsQuery;
+import models.internal.impl.DefaultQueryResult;
+import models.internal.scheduling.Job;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import java.time.ZonedDateTime;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+
+/**
+ * Unit tests for {@link AlertJobRepository}.
+ *
+ * @author Christian Briones (cbriones at dropbox dot com)
+ */
+public class AlertJobRepositoryTest {
+    @Mock
+    private AlertRepository _alertRepository;
+    private UUID _id;
+    private Organization _organization;
+    private Alert _alert;
+
+    private AlertJobRepository _jobRepository;
+    private AlertExecutionContext _context;
+
+    @Before
+    public void setUp() {
+        _organization = TestBeanFactory.createOrganization();
+        _id = UUID.randomUUID();
+        _alert = new DefaultAlert.Builder()
+                .setId(_id)
+                .setOrganization(_organization)
+                .setEnabled(true)
+                .setName("TestAlert")
+                .setDescription("Used in a test.")
+                .setQuery(
+                        new DefaultMetricsQuery.Builder()
+                                .setQuery("Query TBD")
+                                .setStart(ZonedDateTime.now())
+                                .setEnd(ZonedDateTime.now())
+                                .build()
```

Comment:
Same as the previous case - this will be changed once the MetricsQuery changes land.